docs/Provisioner-Introduction.md

```diff
@@ -0,0 +1,71 @@
+### Table of Content
+
+- [Introduction to Provisioner](#introduction-to-provisioner)
+  - [Informational](#informational)
+    - [Table of Content](#table-of-content)
+    - [Document history:](#document-history)
+  - [Overview](#overview)
+  - [Operations Performed by Provisioner](#operations-performed-by-provisioner)
+  - [How Provisioner Interacts With Other Components](#how-provisioner-interacts-with-other-components)
+    - [Northbound Interface](#northbound-interface)
+    - [Southbound Interface](#southbound-interface)
+
+
+### Document history:
+
+| Revision | Revision date | Author           | Description |
+| -------- | :-----------: | ---------------- | ----------- |
+| 0.1      | 2020-10-20    | Ivan Tishchenko  | Initial release |
+
+
+## Overview
+
+See [README.md](../README.md).
+
+
+## Operations Performed by Provisioner
+
+CORTX stack goes through certain stages in it's lifecycle.  Below is the diagram which highlights some of these stages, related to and managed by Provisioner.
+
+![lifecycle](diagrams/lifecycle/lifecycle.png)
+
+* Initial Provisioning: [Auto-Deploy-Provisioner-CLI-Commands](https://github.com/Seagate/cortx-prvsnr/wiki/Auto-Deploy-Provisioner-CLI-Commands).
+  * For reference -- manual for VM-based environment: [Auto-Deploy-VM-Provisioner-CLI-Commands](https://github.com/Seagate/cortx-prvsnr/wiki/Auto-Deploy-VM-Provisioner-CLI-Commands).
+* Boxing & Un-Boxing: [Box-Unbox](https://github.com/Seagate/cortx-prvsnr/wiki/Box-Unbox).
+* CSM On-Boarding: [CORTX Management Portal Quick Start](https://github.com/Seagate/cortx-management-portal/blob/main/docs/CortxManagementPortalQuickstart.md), sections Preboarding and Onboarding.
+* Fail-over -- is managed by HA.  No doc available.
+* Node Replacement: [Node-Replacement](https://github.com/Seagate/cortx-prvsnr/wiki/Node-Replacement).
+  * Extra docs by HA (not sure how relevant):
+    [Node-Replacement](https://github.com/Seagate/cortx-ha/wiki/Node-Replacement), and
+    [CORTX-HA-Replace-Node](https://github.com/Seagate/cortx-ha/wiki/Cortx-ha-Replace-Node).
+* SW Update: [Software-Updates](https://github.com/Seagate/cortx-prvsnr/wiki/Software-Updates).
+
+
+## How Provisioner Interacts With Other Components
+
+### Northbound Interface
+
+Provisioner exposes an API which other components may use to run certain operations through Provisioner.
+
+Main page: [Northbound-Interface](https://github.com/Seagate/cortx-prvsnr/wiki/Northbound-Interface).
+
+This API is exposed in 3 forms:
+
+* Library (python module).
+* CLI (main entry point is command-line utility called `provisioner`; run `provisioner --help` for list of available commands).
+* CLI-wrapper library.  It's a python module, but it's "decoupled" from main library listed above, and does not change frequently with versions which allows for special strict linkage of this python module, e.g. in a form of python frozen app (and there is no need to re-link every time there is new version of main Provisioner library).
```

Comment:
@t7ko-seagate I think we shouldn't emphasize the stability of the API here since it actually doesn't follow any specific rules here: if API needs to be changed we do that. The main rule is to keep it in-sync with underlying parts (cli and lowest level python API).
In the current state the main reason of that 3rd level stability (and random updates) is that it exposes not all commands available for lower levels since:
- the main consumer (CSM) doesn't require them for now
- while updating that level usually requires just few lines of code it will lead to additional testing and for sure some specific bugs might appear due to additional shell-related layer